src/dropck.md

```diff
@@ -22,79 +22,108 @@ let y;
 }
 ```
 
-Each creates its own scope, clearly establishing that one drops before the
-other. However, what if we do the following?
+There are some more complex situations, but the order is still obeyed ‒
+variables are dropped in the reverse order of their definition.
 
-```rust,ignore
-let (x, y) = (vec![], vec![]);
+```rust
+# use std::cell::RefCell;
+struct PushOrder<'a>(&'a RefCell<Vec<usize>>, usize);
+
+impl Drop for PushOrder<'_> {
+    fn drop(&mut self) {
+        self.0.borrow_mut().push(self.1);
+    }
+}
+
+let order = RefCell::new(vec![]);
+{
+    let (a, b);
+    let (c, d) = (PushOrder(&order, 2), PushOrder(&order, 1));
+    a = PushOrder(&order, 4);
+    b = PushOrder(&order, 3);
+}
+assert_eq!(order.into_inner(), vec![1, 2, 3, 4]);
 ```
 
-Does either value strictly outlive the other? The answer is in fact *no*,
-neither value strictly outlives the other. Of course, one of x or y will be
-dropped before the other, but the actual order is not specified. Tuples aren't
-special in this regard; composite structures just don't guarantee their
-destruction order as of Rust 1.0.
+This is not specific to unpacking tuples, it works for any kind of pattern.
 
-We *could* specify this for the fields of built-in composites like tuples and
-structs. However, what about something like Vec? Vec has to manually drop its
-elements via pure-library code. In general, anything that implements Drop has
-a chance to fiddle with its innards during its final death knell. Therefore
-the compiler can't sufficiently reason about the actual destruction order
-of the contents of any type that implements Drop.
+On the other hand, fields inside structs and tuples if they are not unpacked are
+dropped in the order of their definition, as described [here][rfc1857].
+
+```rust
+# use std::cell::RefCell;
+struct PushOrder<'a>(&'a RefCell<Vec<usize>>, usize);
+
+impl Drop for PushOrder<'_> {
+    fn drop(&mut self) {
+        self.0.borrow_mut().push(self.1);
+    }
+}
+
+let order = RefCell::new(vec![]);
+{
+    let tuple = (PushOrder(&order, 2), PushOrder(&order, 1));
+}
+assert_eq!(order.into_inner(), vec![2, 1]);
```

Comment:
IMO it's more clear if you switch the values in the tuple so you can make the assertion be:
```rust
assert_eq!(order.into_inner(), vec![1, 2]);
```
That way as you're reading, you know the first item that gets dropped is the one with the value `1`

Reply:
I wanted to make it clearer it is in the opposite direction than above, but you're right, it looks rather silly if both are in the reversed order.